### PR TITLE
Fix updatedBy field in event stream

### PR DESCRIPTION
**Pull request by aarshi0301**
## Change description

> UpdatedBy field is the delete event is not correct
JIRA : https://atlanhq.atlassian.net/browse/PLT-373

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### Developer Testing Notes:
1. Create Glossary on Beta [with other user profile]
2. Delete Glossary with other user profile
3. Validate Events [updatedBy field should contain userId of step2]
4.  validate auditSearch API

a**. Beta testing**
```
curl --location 'https://beta.atlan.dev/api/meta/entity/auditSearch' \
--header 'authorization: Bearer  your token' \
--header 'content-type: application/json' \
--data '{
    "dsl": {
        "from": 0,
        "size": 10,
        "sort": [
            {
                "created": {
                    "order": "desc"
                }
            }
        ],
        "query": {
            "bool": {
                "filter": {
                    "bool": {
                        "should": [
                        
                            {
                                "term": {
                                    "entityId": "10f0cc44-45c5-449b-bc0e-0b3358989d07"
                                }
                            }
                        ]
                    }
                }
            }
        }
    }
}'
```
```
{
   "entityAudits":[
      {
         "entityQualifiedName":"Wojd826Y8Iw2iz98XdMuM",
         "typeName":"AtlasGlossary",
         "entityId":"10f0cc44-45c5-449b-bc0e-0b3358989d07",
         "timestamp":1701679563047,
         "created":1701692514253,
         "user":"aarshi.arora",
         "action":"ENTITY_DELETE",
         "details":null,
         "eventKey":"10f0cc44-45c5-449b-bc0e-0b3358989d07:1701679563047",
         "entity":null,
         "type":null,
         "detail":{
            "typeName":"AtlasGlossary",
            "attributes":{
               "qualifiedName":"Wojd826Y8Iw2iz98XdMuM"
            },
            "guid":"10f0cc44-45c5-449b-bc0e-0b3358989d07",
            "isIncomplete":false,
            "provenanceType":0,
            "status":"DELETED",
            "createdBy":"ekta.varma",
            "updatedBy":"aarshi.arora",
            "createTime":1701678803307,
            "updateTime":1701679563047,
            "version":0,
            "meanings":[
               
            ],
            "deleteHandler":"DEFAULT",
            "proxy":false
         },
         "entityDetail":{
            "typeName":"AtlasGlossary",
            "attributes":{
               
            },
            "guid":"10f0cc44-45c5-449b-bc0e-0b3358989d07",
            "status":"DELETED",
            "isIncomplete":false,
            "createdBy":"ekta.varma",
            "updatedBy":"aarshi.arora",
            "createTime":1701678803307,
            "updateTime":1701692513730
         },
         "headers":{
            "x-atlan-request-id":"18214413-5553-4c51-3d30-5349daac5bd0",
            "x-atlan-via-ui":"true"
         }
      },
      {
         "entityQualifiedName":"Wojd826Y8Iw2iz98XdMuM",
         "typeName":"AtlasGlossary",
         "entityId":"10f0cc44-45c5-449b-bc0e-0b3358989d07",
         "timestamp":1701678803307,
         "created":1701678803823,
         "user":"ekta.varma",
         "action":"ENTITY_CREATE",
         "details":null,
         "eventKey":"10f0cc44-45c5-449b-bc0e-0b3358989d07:1701678803307",
         "entity":null,
         "type":null,
         "detail":{
            "typeName":"AtlasGlossary",
            "attributes":{
               "popularityScore":1.17549435E-38,
               "assetMcMonitorNames":[
                  
               ],
               "lastSyncRunAt":0,
               "__hasLineage":false,
               "sourceQueryComputeCostRecordList":[
                  
               ],
               "assetSodaLastSyncRunAt":0,
               "starredCount":0,
               "adminUsers":[
                  
               ],
               "lastRowChangedAt":0,
               "sourceReadRecentUserList":[
                  
               ],
               "assetMcIncidentQualifiedNames":[
                  
               ],
               "assetMcIncidentTypes":[
                  
               ],
               "assetSodaLastScanAt":0,
               "sourceUpdatedAt":0,
               "assetDbtJobLastRunArtifactsSaved":false,
               "starredDetailsList":[
                  
               ],
               "isEditable":true,
               "sourceReadCount":0,
               "announcementUpdatedAt":0,
               "sourceCreatedAt":0,
               "assetDbtJobLastRunDequedAt":0,
               "assetDbtTags":[
                  
               ],
               "sourceReadSlowQueryRecordList":[
                  
               ],
               "qualifiedName":"Wojd826Y8Iw2iz98XdMuM",
               "sourceQueryComputeCostList":[
                  
               ],
               "assetDbtJobLastRunNotificationsSent":false,
               "assetMcMonitorTypes":[
                  
               ],
               "assetSodaCheckCount":0,
               "assetMcMonitorStatuses":[
                  
               ],
               "starredBy":[
                  
               ],
               "sourceLastReadAt":0,
               "name":"test",
               "certificateUpdatedAt":1701678803307,
               "assetMcIncidentSeverities":[
                  
               ],
               "sourceReadQueryCost":0.0,
               "ownerUsers":[
                  "ekta.varma"
               ],
               "certificateStatus":"DRAFT",
               "assetDbtJobLastRunHasSourcesGenerated":false,
               "assetMcIncidentSubTypes":[
                  
               ],
               "isAIGenerated":false,
               "assetDbtJobLastRunHasDocsGenerated":false,
               "assetTags":[
                  
               ],
               "assetMcIncidentStates":[
                  
               ],
               "assetDbtJobLastRunUpdatedAt":0,
               "ownerGroups":[
                  
               ],
               "certificateUpdatedBy":"ekta.varma",
               "assetMcMonitorQualifiedNames":[
                  
               ],
               "sourceReadExpensiveQueryRecordList":[
                  
               ],
               "assetDbtJobLastRunStartedAt":0,
               "isDiscoverable":true,
               "isPartial":false,
               "sourceReadTopUserRecordList":[
                  
               ],
               "assetMcMonitorScheduleTypes":[
                  
               ],
               "viewerUsers":[
                  
               ],
               "viewScore":1.17549435E-38,
               "sourceReadTopUserList":[
                  
               ],
               "assetMcIncidentNames":[
                  
               ],
               "userDescription":"",
               "adminRoles":[
                  
               ],
               "adminGroups":[
                  
               ],
               "assetDbtJobLastRunCreatedAt":0,
               "assetDbtJobNextRun":0,
               "sourceReadRecentUserRecordList":[
                  
               ],
               "assetIcon":"atlanGlossary",
               "sourceReadPopularQueryRecordList":[
                  
               ],
               "sourceTotalCost":0.0,
               "assetMcLastSyncRunAt":0,
               "sourceReadUserCount":0,
               "viewerGroups":[
                  
               ],
               "assetDbtJobLastRun":0
            },
            "guid":"10f0cc44-45c5-449b-bc0e-0b3358989d07",
            "isIncomplete":false,
            "provenanceType":0,
            "createdBy":"ekta.varma",
            "updatedBy":"ekta.varma",
            "createTime":1701678803307,
            "updateTime":1701678803307,
            "version":0,
            "classifications":[
               
            ],
            "proxy":false
         },
         "entityDetail":{
            "typeName":"AtlasGlossary",
            "attributes":{
               
            },
            "guid":"10f0cc44-45c5-449b-bc0e-0b3358989d07",
            "status":"DELETED",
            "isIncomplete":false,
            "createdBy":"ekta.varma",
            "updatedBy":"aarshi.arora",
            "createTime":1701678803307,
            "updateTime":1701692513730
         },
         "headers":{
            "x-atlan-request-id":"c4977a17-2d96-4264-38ff-4c6509b299f6",
            "x-atlan-via-ui":"true"
         }
      }
   ],
   "aggregations":null,
   "count":2,
   "totalCount":2
}
```

b. **staging testing** :
```
curl --location 'https://staging-governance.atlan.dev/api/meta/entity/auditSearch' \
--header 'content-type: application/json' \
--header 'x-atlan-request-id: bdd9c786-7861-4672-b4fd-6b7779dfc7bd' \
--header 'Authorization: Bearer your auth token' \
--data '{
    "dsl": {
        "from": 0,
        "size": 10,
        "sort": [
            {
                "created": {
                    "order": "desc"
                }
            }
        ],
        "query": {
            "bool": {
                "filter": {
                    "bool": {
                        "should": [
                        
                            {
                                "term": {
                                    "entityId": "03e5f988-fd06-489f-9e14-63a1af8b51f6"
                                }
                            }
                        ]
                    }
                }
            }
        }
    }
}'
```

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java [hunk 1]
```diff
@@ -1917,6 +1917,7 @@ private EntityMutationResponse deleteVertices(Collection<AtlasVertex> deletionCa
                 }
                 entity.setDeleteHandler(handler);
                 entity.setStatus(Status.DELETED);
+                entity.setUpdatedBy(RequestContext.get().getUser());
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java:1920]: What will be the value for updateBy attribute in Entity audits? Make sure both are same

Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java:1920]: Also, do we have information on what all subscribers are listening to ATLAS_ENTITIES?
Make sure we do not break any existing listeners listening to these type of notifications if any

Comment (aarshi0301) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasEntityStoreV2.java:1920]: Hi @nikhilbonte21 
`deleteVertices` method just updates entity attributes which are later consumed by both audits and notification listeners.
In the above case, the  username who actually deleted the glossary will be used in `updatedBy` field.
Attached screenshot for your reference.
Screenshot 2023-12-06 at 12 08 41 PM

**Review by n5nk — APPROVED** [review]
LGTM

**Review by checkaayush — APPROVED** [review]

**Review by nikhilbonte21 — APPROVED** [review]

**Review by sumandas0 — APPROVED** [review]

**Pull request merged**